### DISCUSSION TITLE
Plugin/Extension idea

### DISCUSSION THREAD

**Discussion by Andre601** [mkdocs/mkdocs#2690] [Ideas]
There is an idea I have that could be made into a plugin or extension for MkDocs. Unfortunately do I not have the knowledge to actually code such an extension myself.

The idea is (probably) simple: A plugin/extension that would integrate [giscus](https://giscus.app) as a comment-system into the mkdocs pages based on settings you provide through its configuration, optionally with a per-site override through meta.

The reason why I suggest this is, that while giscus is a nice system to use, things such as its configuration are something that can't easily be done through simple configuration and - as of right now - requires theme extension and such to achieve this.

**Comment by simondueckert** [comment]
Can giscus be integrated in the meantime? I have documentation based on material theme where this tutorial works (https://squidfunk.github.io/mkdocs-material/setup/adding-a-comment-system).

How can I add giscus when I use the readthedocs theme?

**Reply by Andre601** [reply]
Probably the same way you would in material for MkDocs: By extending the theme and editing whatever file is used for the page content itself (Usually one called main.html or base.html)

**Comment by simondueckert** [comment]
Currently for material theme I have a overrides/partials/comment.html file for the embedding of giscus. So you mean I have to rename comment.html to main.html (or base.html)?

**Reply by Andre601** [reply]
comment.html is just a separate file that material for MkDocs includes in the main html file through Jinja2's `include` feature.
So it's basically like copy-pasting, yeah.

(Also, you should reply to the post I made, so that it remainds properly structured, instead of making a separate reply here.)
